app/src/main/java/com/github/onlynotesswent/model/ocr/TextRecognizer.kt

```diff
@@ -0,0 +1,106 @@
+package com.github.onlynotesswent.model.ocr
+
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import java.io.IOException
+
+/**
+ * TextRecognizer class that extracts text from an image URI using Google ML Kit’s Text Recognition
+ * API.
+ *
+ * @param activity the ComponentActivity that will use the TextRecognizer
+ */
+class TextRecognizer(private val activity: ComponentActivity) {
+  private lateinit var textRecognitionLauncher: ActivityResultLauncher<String>
+  private val textRecognizer = TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
```

Comment:
Try including with a default value as a class parameter, to enable easier testing and possibility of initialization with other options